ipa-core/src/protocol/ipa_prf/aggregation/breakdown_reveal.rs

```diff
@@ -24,18 +25,74 @@ use crate::{
             },
             oprf_padding::{apply_dp_padding, PaddingParameters},
             prf_sharding::{AttributionOutputs, SecretSharedAttributionOutputs},
-            shuffle::{shuffle_attribution_outputs, Shuffle},
+            shuffle::{Shuffle, Shuffleable},
             BreakdownKey,
         },
         BooleanProtocols, RecordId,
     },
     secret_sharing::{
-        replicated::semi_honest::AdditiveShare as Replicated, BitDecomposed, FieldSimd,
-        TransposeFrom, Vectorizable,
+        replicated::{semi_honest::AdditiveShare as Replicated, ReplicatedSecretSharing},
+        BitDecomposed, FieldSimd, SharedValue, TransposeFrom, Vectorizable,
     },
     seq_join::seq_join,
 };
 
+impl<BK, TV> AttributionOutputs<Replicated<BK>, Replicated<TV>>
+where
+    BK: BooleanArray,
+    TV: BooleanArray,
+{
+    fn join_fields(breakdown_key: BK, trigger_value: TV) -> <Self as Shuffleable>::Share {
+        let mut share = <Self as Shuffleable>::Share::ZERO;
+
+        BooleanArrayWriter::new(&mut share)
+            .write(&breakdown_key)
+            .write(&trigger_value);
+
+        share
+    }
+
+    fn split_fields(share: &<Self as Shuffleable>::Share) -> (BK, TV) {
+        let bits = BooleanArrayReader::new(share);
+        let (breakdown_key, bits) = bits.read();
+        let (trigger_value, _bits) = bits.read();
```

Comment:
possibly dumb question, but how does this know how many bits to read? (more practically, how would I do the same thing for the new type I'm working with in the hybrid protocol?)

Reply:
It is determined by the type parameter (i.e. `T` is `BK` and `BK::BITS` is 5 or 8). The type parameter in turn is determined by type inference working backwards from the return value.

Reply:
👨‍🍳 😽